pkg/cli/upgradeassistant/internal/repository/mongodb/180/project.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2021 The KodeRover Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package _80
+
+import (
+	"context"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	_80 "github.com/koderover/zadig/pkg/cli/upgradeassistant/internal/repository/models/180"
+	"github.com/koderover/zadig/pkg/microservice/aslan/config"
+	mongotool "github.com/koderover/zadig/pkg/tool/mongo"
+)
+
+type ProductColl struct {
+	*mongo.Collection
+
+	coll string
+}
+
+func NewProductColl() *ProductColl {
+	name := _80.Project{}.TableName()
+	return &ProductColl{Collection: mongotool.Database(config.MongoDatabase()).Collection(name), coll: name}
+}
+
+func (c *ProductColl) GetCollectionName() string {
+	return c.coll
+}
+
+func (c *ProductColl) EnsureIndex(ctx context.Context) error {
+	mod := mongo.IndexModel{
+		Keys:    bson.M{"product_name": 1},
+		Options: options.Index().SetUnique(true),
+	}
+
+	_, err := c.Indexes().CreateOne(ctx, mod)
+
+	return err
+}
+
+func (c *ProductColl) List() ([]*_80.Project, error) {
+	var resp []*_80.Project
+
+	cursor, err := c.Collection.Find(context.TODO(), bson.M{})
```

Comment:
will c.Collection.Find return error when no document is found? If so ua will panic, this situation shoule be considered.